closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -580,17 +618,18 @@ int decryptFinal(ByteBuffer src, ByteBuffer dst)
 
     // return tag length in bytes
     int getTagLen() {
-
         return this.tagLenBytes;
     }
 
     int getBufferedLength() {
-        if (ibuffer != null && decrypting) {
-            return ibuffer.size();
-        }
+        synchronized(this) {
+            if (ibuffer != null && decrypting) {
+                return ibuffer.size();
+            }
 
-        if (ibuffer_enc != null && !decrypting) {
-            return ibuffer_enc.size();
+            if (ibuffer_enc != null && !decrypting) {
+                return ibuffer_enc.size();
+            }
```

Comment:
This is harder to read than necessary; suggest:
```
            if (decrypting) {
                if (ibuffer != null) {
                    return ibuffer.size();
                }
            } else {
                if (ibuffer_enc != null) {
                    return ibuffer_enc.size();
                }
            }
```

Reply:
Fixed.